src/main/java/co/cask/hydrator/plugin/batch/DynamoDBBatchSource.java

```diff
@@ -0,0 +1,220 @@
+      "by a pipe sign '|' and specifies the tokens and actual attribute names. For example, '#yr|year', if the query " +
+      "is like: '#yr = :yyyy'. This might be necessary if an attribute name conflicts with a DynamoDB reserved word. " +
+      "(Macro Enabled)")
+    @Nullable
+    @Macro
+    private String nameMappings;
+    @Description("List of the placeholder tokens used as the attribute values in the 'Query or FilterQuery' along " +
+      "with their actual values. This is a comma-separated list of key-value pairs, where each pair is separated by a" +
+      " pipe sign '|' and specifies the tokens and actual values. For example, ':v_id|256,:v_title|B', if the query " +
+      "is like: 'ID = :v_id AND Title = :v_title'. (Macro Enabled)")
+    @Macro
+    private String valueMappings;
+    @Description("List of the placeholder tokens used as the attribute values in the 'Query or FilterQuery' along " +
+      "with their data types. This is a comma-separated list of key-value pairs, where each pair is separated by a " +
+      "pipe sign '|' and specifies the tokens and its type. For example, ':v_id|int,:v_title|string', if the query is" +
+      " like: 'ID = :v_id AND Title = :v_title'. Supported types are: 'boolean, int, long, float, double and string'." +
+      " (Macro Enabled)")
+    @Macro
+    private String placeholderType;
+
+    @Description("Read Throughput for AWS DynamoDB table to connect to, in double. Default is 1. (Macro Enabled)")
+    @Nullable
+    @Macro
+    private String readThroughput;
+
+    @Description("Read Throughput Percentage for AWS DynamoDB table to connect to. Default is 0.5. (Macro Enabled)")
+    @Nullable
+    @Macro
+    private String readThroughputPercentage;
+
+    public DynamoDBConfig(String referenceName, String accessKey, String secretAccessKey, @Nullable String regionId,
+                          @Nullable String endpointUrl, String tableName, String query, @Nullable String filterQuery,
+                          @Nullable String nameMappings, String valueMappings, String placeholderType,
+                          @Nullable String readThroughput, @Nullable String readThroughputPercentage) {
+      super(referenceName);
+      this.accessKey = accessKey;
+      this.secretAccessKey = secretAccessKey;
+      this.regionId = regionId;
+      this.endpointUrl = endpointUrl;
+      this.tableName = tableName;
+      this.query = query;
+      this.filterQuery = filterQuery;
+      this.nameMappings = nameMappings;
+      this.valueMappings = valueMappings;
+      this.placeholderType = placeholderType;
+      this.readThroughput = readThroughput;
+      this.readThroughputPercentage = readThroughputPercentage;
+    }
+
+    /**
+     * Validates whether the table name follows the DynamoDB naming rules and conventions or not.
+     */
+    private void validateTableName() {
```

Comment:
Currently, tableName is not macro enabled. Hence, kept this validation in configurePipleine() only.

We will add validation on prepareRun() also, once we enabled the macro for tableName.